addons/web/static/src/views/fields/properties/property_value.js

```diff
@@ -0,0 +1,290 @@
+        } else if (this.props.type === 'many2one') {
+            if (!value || value.length !== 2 || !value[0]) {
+                return false;
+            }
+            return value;
+        } else if (this.props.type === 'many2many') {
+            if (!value || !value.length) {
+                return [];
+            }
+
+            // Convert to TagList format
+            return value.map((many2manyValue) => {
+                return {
+                    'id': many2manyValue[0],
+                    'text': many2manyValue[1],
+                    'onClick': () => this._openRecord(this.props.model, many2manyValue[0]),
+                    'onDelete': !this.props.readonly && (() => this.onMany2manyDelete(many2manyValue[0])),
+                    'colorIndex': 0,
+                };
+            });
+        } else if (this.props.type === 'tags') {
+            return value || [];
+        }
+
+        return value;
+    }
+
+    /**
+     * Return the model domain (related to many2one and many2many properties).
+     */
+    get propertyDomain() {
+        if (!this.props.domain || !this.props.domain.length) {
+            return [];
+        }
+        return new Domain(this.props.domain).toList();
+    }
+
+    /**
+     * Formatted value displayed in readonly mode.
+     */
+    get displayValue() {
+        const value = this.propertyValue;
+
+        if (this.props.type === 'many2one' && value && value.length === 2) {
+            return `${value[1]} (#${value[0]})`;
+        } else if (!value) {
+            return false;
+        } else if (this.props.type === 'datetime' && value) {
+            return value.toFormat('LL/dd/yyyy HH:mm:ss');
+        } else if (this.props.type === 'date' && value) {
+            return value.toFormat('LL/dd/yyyy');
+        } else if (this.props.type === 'selection') {
+            return this.props.selection.find(option => option[0] === value)[1];
+        }
+        return value.toString();
+    }
+
+    /**
+     * Open the form view of the given record id / model.
+     */
+    _openRecord(recordModel, recordId) {
+        this.orm.call(
+            recordModel,
+            'get_formview_action',
+            [[recordId]],
+            { context: this.props.context },
+        ).then((action) => {
+            this.action.doAction(action);
+        });
+    }
+
+    /**
+     * Parse the value received by the sub-components and trigger an onChange event.
+     */
+    async onValueChange(newValue) {
+        if (this.props.type === 'datetime') {
+            if (typeof newValue === 'string') {
+                newValue = DateTime.fromISO(newValue);
+            }
+            newValue = newValue.toUTC().toFormat(DATETIME_FORMAT);
+        } else if (this.props.type === 'date') {
+            if (typeof newValue === 'string') {
+                newValue = DateTime.fromISO(newValue);
+            }
+            newValue = newValue.toFormat(DATE_FORMAT);
```

Comment:
I'm guessing this is all timezone compliant with locale defined in user settings, and not the one from the browser?